lib/taskana-core/src/main/java/pro/taskana/common/api/TaskanaEngine.java

```diff
@@ -16,131 +16,168 @@
+   * Checks if the {@linkplain pro.taskana.spi.history.api.TaskanaHistory TaskanaHistory} plugin is
+   * enabled.
    *
-   * @return true if the history is enabled. Otherwise false.
+   * @return true if the history is enabled; otherwise false
    */
   boolean isHistoryEnabled();
 
   /**
-   * gets the current connection management mode.
+   * Returns the {@linkplain ConnectionManagementMode ConnectionManagementMode} of the
+   * TaskanaEngine.
    *
-   * @return the current connection management mode.
+   * @return {@linkplain ConnectionManagementMode ConnectionManagementMode}
    */
   ConnectionManagementMode getConnectionManagementMode();
 
   /**
-   * sets the connection management mode.
+   * Sets {@linkplain ConnectionManagementMode ConnectionManagementMode} of the TaskanaEngine.
    *
-   * @param mode the connection management mode Valid values are:
+   * @param mode the valid values for the {@linkplain ConnectionManagementMode} are:
    *     <ul>
-   *       <li>PARTICIPATE - taskana participates in global transaction. This is the default mode.
-   *       <li>AUTOCOMMIT - taskana commits each API call separately
-   *       <li>EXPLICIT - commit processing is managed explicitly by the client
+   *       <li>{@linkplain ConnectionManagementMode#PARTICIPATE PARTICIPATE} - taskana participates
+   *           in global transaction; this is the default mode
+   *       <li>{@linkplain ConnectionManagementMode#AUTOCOMMIT AUTOCOMMIT} - taskana commits each
+   *           API call separately
+   *       <li>{@linkplain ConnectionManagementMode#EXPLICIT EXPLICIT} - commit processing is
+   *           managed explicitly by the client
    *     </ul>
    */
   void setConnectionManagementMode(ConnectionManagementMode mode);
 
   /**
-   * Set the connection to be used by TASKANA in mode {@linkplain
-   * ConnectionManagementMode#EXPLICIT}. If this Api is called, taskana uses the connection passed
-   * by the client for all subsequent API calls until the client resets this connection. Control
-   * over commit and rollback of the connection is the responsibility of the client. In order to
-   * close the connection, closeConnection() or setConnection(null) has to be called.
+   * Set the {@code Connection} to be used by TASKANA in mode {@linkplain
+   * ConnectionManagementMode#EXPLICIT EXPLICIT}. If this API is called, TASKANA uses the {@code
+   * Connection} passed by the client for all subsequent API calls until the client resets this
+   * {@code Connection}. Control over commit and rollback of the {@code Connection} is the
+   * responsibility of the client. In order to close the {@code Connection}, {@code
+   * closeConnection()} or {@code setConnection(null)} has to be called.
    *
-   * @param connection - The java.sql.Connection that is controlled by the client
+   * @param connection - The {@code java.sql.Connection} that is controlled by the client
    * @throws SQLException if a database access error occurs
    */
```

Comment:
why do you use `@code` here. should we link that aswell?

Reply:
According to the guide, we don't link external classes. I used @code because I saw it being used somewhere else and didn't think much. Should we just write java.sql.Connection instead? Do do write Connection instead of {@code Connection}?

Reply:
good questions. -> separate ticket.